recipes/Auto_Documentation/Auto_Documentation.ipynb

```diff
@@ -0,0 +1,303 @@
+   ],
+   "metadata": {
+    "id": "5d0sWaZ7YLHN"
+   }
+  },
+  {
+   "cell_type": "code",
+   "source": [
+    "import requests\n",
+    "from time import sleep\n",
+    "\n",
+    "def get_github_repo_contents(repo_url, directory_path, github_token = None):\n",
+    "    # Extracting the owner and repo name from the URL\n",
+    "    repo_parts = repo_url.rstrip('/').split('/')\n",
+    "    owner = repo_parts[-2]\n",
+    "    repo = repo_parts[-1]\n",
+    "\n",
+    "    api_url = f\"https://api.github.com/repos/{owner}/{repo}/contents/{directory_path}\"\n",
+    "    if github_token is not None:\n",
+    "      headers = {'Authorization': f'token {github_token}'}\n",
+    "      response = requests.get(api_url, headers = headers)\n",
+    "    else:\n",
+    "      response = requests.get(api_url)\n",
+    "    response.raise_for_status()\n",
+    "\n",
+    "    contents = response.json()\n",
+    "\n",
+    "    result = []\n",
+    "\n",
+    "    for item in contents:\n",
+    "        if item['type'] == 'file':\n",
+    "            file_response = requests.get(item['download_url'])\n",
+    "            file_response.raise_for_status()\n",
+    "            file_content = file_response.text\n",
+    "            language = item['name'].split('.')[-1]\n",
+    "            if language == 'py':\n",
+    "                language = 'python'\n",
+    "            elif language == 'js':\n",
+    "                language = 'javascript'\n",
+    "            result.append(f\"{item['path']}\\n```{language}\\n{file_content}\\n```\")\n",
+    "        elif item['type'] == 'dir':\n",
+    "            # Recursively go through subdirectories\n",
+    "            subdirectory_contents = get_github_repo_contents(repo_url, item['path'], github_token)\n",
+    "            result.append(subdirectory_contents)\n",
+    "        sleep(0.1)\n",
+    "\n",
+    "    return \"\\n\\n\".join(result)\n"
+   ],
+   "metadata": {
+    "id": "3JFi40LArpIa"
+   },
+   "execution_count": null,
+   "outputs": []
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "## Get code from `ibm-granite-community/utils`\n",
+    "\n",
+    "In this example, we're focusing on the `ibm-granite-community/utils` repository, specifically the `ibm_granite_community` directory. This directory contains various utility functions that we want to document.\n",
+    "\n",
+    "By specifying this directory, we ensure that we're only fetching the relevant code and not unnecessary files or directories. This helps to keep our input focused and reduces the likelihood of exceeding token limits in our AI model."
+   ],
+   "metadata": {
+    "id": "H-06VQn1YmtU"
+   }
+  },
+  {
+   "cell_type": "code",
+   "source": [
+    "prompt = get_github_repo_contents(\"https://github.com/ibm-granite-community/utils\", \"ibm_granite_community\")"
```

Comment:
You might want to use `granite-kitchen` for this? The utils project will only really have `get_env_var()` in it.

Reply:
Ahh, now I see. Candidly, I hate to import another library, just to save three lines of code.

Reply:
I mean document `granite-kitchen` instead of `utils`. I've just removed most of the code from `utils`.

Reply:
OK, looking at this now, yes, I want utils, not granite kitchen. I'm trying to document functions and methods, kitchen is all notebooks. Thank you for clearing them out of utils they muddled this example.